sdk/webpubsub/Azure.Messaging.WebPubSub.Client/src/EventSource/WebPubSubClientEventSource.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Text;
+using Azure.Core.Diagnostics;
+
+namespace Azure.Messaging.WebPubSub.Clients
+{
+    [EventSource(Name = EventSourceName)]
+    internal class WebPubSubClientEventSource: AzureEventSource
+    {
+        private const string EventSourceName = "Azure-Messaging-WebPubSub-Client";
+
+        private WebPubSubClientEventSource() : base(EventSourceName)
+        {
+        }
+
+        // Having event ids defined as const makes it easy to keep track of them
+        private const int ClientStartingId = 1;
+        private const int ClientStateChangedId = 2;
+        private const int WebSocketConnectingId = 3;
+        private const int WebSocketClosedId = 4;
+        private const int FailedToProcessMessageId = 5;
+        private const int FailedToReceiveBytesId = 6;
+        private const int FailedToChangeClientStateId = 7;
+        private const int StopRecoveryId = 8;
+        private const int FailedToRecoverConnectionId = 9;
+        private const int FailedToInvokeEventId = 10;
+        private const int ConnectionConnectedId = 11;
+        private const int ConnectionDisconnectedId = 12;
+        private const int FailedToReconnectId = 13;
+
+        public static WebPubSubClientEventSource Log { get; } = new WebPubSubClientEventSource();
+
+        [Event(1, Level = EventLevel.Verbose, Message = "Client starting.")]
```

Comment:
Shall we make connection start/stop to __Information__ level cause it's kind of stability.